src/app/models/record-vo.ts

```diff
@@ -155,6 +156,20 @@ export class RecordVO
       this.ShareVOs = this.ShareVOs.map((data) => new ShareVO(data));
     }
   }
+
+  public getDownloadOptionsList() {
+    const files = [...(this.FileVOs ?? [])];
+    return files
+      .sort(
+        (a, b) =>
+          Number(b.format === FileFormat.Original) -
+          Number(a.format === FileFormat.Original),
```

Comment:
I ended up using this same pattern in this method. Again, despite its cursed appearance I think this is more flexible than previous implementations method for getting the original file, which tries to find it manually and copy it to the top of the array.